ast/ast.hpp

```diff
@@ -131,19 +134,45 @@ struct RecordType : Type {
     void accept(Visitor& v) override { v.visit(this); }
 };
 struct Variable : Node {
-    void accept(Visitor& v) override { v.visit(this); }
+    lexer::Token name;
+    sPtr<Type> type;
+    sPtr<Expression> expression;
+    bool operator==(const Variable& other) const {
+        return Node::operator==(other)
+                && name == other.name && type == other.type;
+    }
+    void accept(Visitor& v) override {
+        v.visit(this);
+    }
 };
 struct Body : Node {
-    void accept(Visitor& v) override { v.visit(this); }
+    std::vector<sPtr<Statement>> statements;
+    std::vector<sPtr<Variable>> variables;
+    std::vector<sPtr<Type>> types;
+    bool operator==(const Body& other) const {
+        return Node::operator==(other)
+                && types == other.types && variables == other.variables && statements == other.statements;
+    }
+    void accept(Visitor& v) override {
+        v.visit(this);
+    }
 };
 struct Statement : Node {
     void accept(Visitor& v) override { v.visit(this); }
 };
 struct Assignment : Statement {
-    void accept(Visitor& v) override { v.visit(this); }
+    sPtr<Expression> LeftExpression;
+    sPtr<Expression> RightExpression;
```

Comment:
Don't start a variable's name with a capital letter. The convention here is to use `lhs`/`rhs`, or even `lvalue`/`rvalue` when it comes to assignment